packages/react-components/src/components/ParticipantsButton.tsx

```diff
@@ -0,0 +1,209 @@
+ *
+ * @param props - of type ParticipantsButtonProps
+ */
+export const ParticipantsButton = (props: ParticipantsButtonProps): JSX.Element => {
+  const {
+    participantListProps,
+    showLabel = false,
+    callInvitationURL,
+    styles,
+    onMuteAll,
+    onRenderIcon,
+    onRenderText
+  } = props;
+
+  participantListProps.excludeMe = true;
+
+  const onMuteAllCallback = useCallback(() => {
+    if (onMuteAll) {
+      onMuteAll();
+    }
+  }, [onMuteAll]);
+
+  const onRenderCallback = useCallback(() => {
+    return (
+      <Stack className={mergeStyles(defaultParticipantListContainerStyle, styles?.participantListContainerStyle)}>
+        <ParticipantList {...participantListProps} />
+      </Stack>
+    );
+  }, [participantListProps, participantListProps.participants, participantListProps?.myUserId, styles]);
+
+  const onCopyCallback = useCallback(() => {
+    if (callInvitationURL) {
+      return copy(callInvitationURL);
+    }
+    return false;
+  }, [callInvitationURL]);
+
+  const generateDefaultRemoteParticipantsSubMenuProps = (): IContextualMenuItem[] => {
+    const items: IContextualMenuItem[] = [];
+
+    if (participantListProps.participants.length > 0) {
+      items.push({
+        key: 'remoteParticipantsKey',
+        text: 'Remote Participant list',
+        onRender: onRenderCallback
+      });
+
+      items.push({ key: 'participantsDivider1', itemType: ContextualMenuItemType.Divider });
+
+      if (onMuteAll) {
+        items.push({
+          key: 'muteAllKey',
+          text: 'Mute all',
+          title: 'Mute all',
+          iconProps: { iconName: 'MicOff2' },
+          onClick: onMuteAllCallback
+        });
+      }
+    }
+
+    return items;
+  };
+
+  const generateDefaultMenuProps = (): IContextualMenuProps => {
+    const defaultMenuProps: IContextualMenuProps = {
+      items: []
+    };
+
+    if (participantListProps.participants.length > 0) {
+      const participantIds = participantListProps.participants.map((p) => p.userId);
+      let participantCount = participantIds.length;
+
+      if (participantListProps.myUserId && participantIds.indexOf(participantListProps.myUserId) !== -1) {
+        participantCount -= 1;
+        defaultMenuProps.items.push({
+          key: 'selfParticipantKey',
+          name: 'In this call'
+        });
+      }
+
+      defaultMenuProps.items.push({
+        key: 'remoteParticipantCountKey',
+        name: `${participantCount} people`,
+        iconProps: { iconName: 'People' },
+        subMenuProps: {
+          items: generateDefaultRemoteParticipantsSubMenuProps()
+        }
+      });
+    }
+
+    if (callInvitationURL) {
+      defaultMenuProps.items.push({
+        key: 'InviteLinkKey',
+        name: 'Copy invite link',
+        iconProps: { iconName: 'Link' },
+        onClick: onCopyCallback
+      });
+    }
+
+    return defaultMenuProps;
+  };
+
+  const defaultMenuProps = useMemo(() => {
```

Comment:
Is eslint able to warn you if you drop one of the dependencies from the list here?

If not, consider extracting `generateDefaultMenyProps` and it's helpers as top-level private functions in this file.
That way, you have to pass in arguments instead of closure capture and eslint should be able to point out any you miss.